LiskHQ/lisk-sdk
# Add incentivizing Lisk-BFT Protocol Participation - Closes #4354

### What was the problem?
`Incentivizing Lisk-BFT Protocol Participation` section of the [LIP-0014](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0014.md#incentivizing-lisk-bft-protocol-participation) was not implemented

### How did I solve it?
I've created a method in BFT module called `isBFTProtocolCompliant` which returns whether or not the block follows the BFT protocol.
This method is used in the block processor validation pipeline in order to check if the reward was reduced to 25% if the protocol is not followed and it's also used in the block processor create pipeline in order to reduce the reward if the protocol is not followed.

### How to manually test it?
Forge a block with one of the conditions to make the block not follow the BFT protocol and check the reward value

### Review checklist

- [ ] The PR resolves #4354
